### DISCUSSION TITLE
RaspberryMatic 3.63.9.20220430

### DISCUSSION THREAD

**Discussion by jens-maus** [OpenCCU/OpenCCU#1818] [Announcements]
This is release 3.63.9.20220430 of RaspberryMatic which is a **maintenance** release with the following bugfixes and feature changes:

[![Downloads](https://img.shields.io/github/downloads/jens-maus/RaspberryMatic/3.63.9.20220430/total.svg?style=flat-square)](https://github.com/jens-maus/RaspberryMatic/releases/3.63.9.20220430) ![License](https://img.shields.io/github/license/jens-maus/RaspberryMatic.svg?style=flat-square) [![Donate](https://img.shields.io/badge/donate-PayPal-green.svg?style=flat-square)](https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=RAQSDY9YNZVCL) [![GitHub sponsors](https://img.shields.io/static/v1?label=Sponsor&message=%E2%9D%A4&logo=GitHub&link=https://github.com/sponsors/jens-maus)](https://github.com/sponsors/jens-maus)

<sub>[Hier klicken für deutschsprachige 🇩🇪🇦🇹🇨🇭 Version des ChangeLogs/Diskussionsbeitrag](https://homematic-forum.de/forum/viewtopic.php?f=65&t=XXXXX)</sub>

## :construction: Changes:
<sub>For all changes, see the [full commit log](https://github.com/jens-maus/RaspberryMatic/compare/3.63.8.20220330...3.63.9.20220430).</sub>

#### CCU/HomeMatic service changes:
- updated [OCCU](https://github.com/eq-3/occu) version to [3.63.9-1](https://github.com/jens-maus/occu/tree/b_3_63) integrating full compatibility to the [CCU3 3.63.9 firmware](https://www.eq-3.de/downloads/software/firmware/ccu3-firmware/CCU3-Changelog.3.63.9.pdf) which comes with the following changes:
  - Updated the `RPI-RF-MOD` co-processor firmware to version 4.4.22 which should fix an issue where rf communication with certain HomeMatic/BidCos-RF devices were not working reliable (https://homematic-forum.de/forum/viewtopic.php?f=27&t=71025).
  - The `ELV-SH-WUA` device was not correctly recognized in the WebUI.
  - The device config pages for the `HmIP-FALMOT` was not correctly shown.
- added a new `/bin/checkCoProcessor.sh` check script and put it to the global monitrc to be executed regularly for performing regular rf-module/co-processor connectivity checks. In case of identifying a failed rf-module connection a WebUI alarm message will now be raised (https://homematic-forum.de/forum/viewtopic.php?f=65&t=73697).
- added a new experimental `/usr/bin/checkPortForwarding.sh` check script which can be used to try to identify any security critical port forwarding being enabled in the internet router where RaspberryMatic is being connected to.
- updated Mediola NEO SERVER to latest 2.9.0 package version.

#### WebUI changes:
- added new `0168-WebUI-Fix-HmipChannelDialogsInclude` WebUI bugfix patch which adds a missing source include statement which fixes a WebUI crash bug in the new `BOILER_TRANSMITTER` functionality of 3.63.8. This should make the device config settings page for the HmIP-FALMOT-C12 accessible again.
- added a package patch `0001-default-identity-encoding.patch` to the tcl package to make the "identity" character encoding the default+system encoding like it had been with tclsh 8.2 and which seem to have caused the umlaut character encoding issues recently seen. In addition, adapted tclrega and tclrpc accordingly so that they only actively perform character conversion in case no "identity" encoding is used. Furthermore, removed the explicit "convertto" char conversion calls in `jsonrpc.tcl` so fix similar issues. This altogether fixes the char encoding issues seen in the WebUI and external CCU addons like hm_pdetect, etc. (#1782).
- added a tcl package patch which will make sure that the `tcl_precision` will be set to 12 as a default value like this was the case with the older tcl 8.2 version. This should fix recently appearing floating point arithmetic issues which resulted in incorrect valve position calculations in the WebUI (#1784).
- disabled the forced restart of `HMIPServer` upon changing the log level. Now a user has to perform a manual reboot or restart of `HMIPServer` in case he wants to have the log level changes to be applied (#1742).

#### Operating system changes:
- updated all `rpi` and `tinkerboard` kernel defconfigs to not use `CONFIG_SLUB_DEBUG` anymore because we are now exclusively using the newer `init_on_alloc`/`init_on_free` instead of slub debug. Also added the missing `CONFIG_ARM_MODULE_PLTS` config option to the `tinkerboard` defconfig to fix various "relocation 28 out of range" errors after the 5.15.x kernel update (#1435).
- synchronized all used kernel options throughout our supported platforms.
- updated buildroot to latest 2022.02.1 version.
- added `CONFIG_AUDIT` to all kernel defconfigs to enable kernel auditing support so that the kernels spits out audit messages in case invalid syscalls are being processed.
- updated `wiringpi-odroid` package to latest version available.
- added direct USB storage quirks for the RaspberryPi target so that some USB<>SSD connectors will start to work out-of-the-box (#1727, https://github.com/home-assistant/operating-system/pull/1864).
- bumped all Linux kernel versions for all our supported platforms from 5.10.x to 5.15.x and used the stable 1.20220331 (5.15.32) version for the RaspberryPi platform and the upstream 5.15.53 LTS version for the other platforms.

## :family: Contributors (alphabetically):
- @jens-maus

## :memo: Support:
For support on installation and help please visit the following web pages:

[RaspberryMatic - Documentation :us:](https://github.com/jens-maus/RaspberryMatic/wiki/en.Home)
[RaspberryMatic - Discussions :us:](https://github.com/jens-maus/RaspberryMatic/discussions)
[RaspberryMatic - Dokumentation :de:](https://github.com/jens-maus/RaspberryMatic/wiki)
[RaspberryMatic - Forum :de:](https://homematic-forum.de/forum/viewforum.php?f=65)

## :package:  Download:
The following installation archives can be downloaded for different hardware platforms. To verify their integrity a `sha256` checksum is provided as well. You can either upload these files using the WebUI-based update mechanism or unarchive them to e.g. flash the included `*.img` files on a fresh installation media (e.g. microSD card):

- RaspberryPi4 Model B, RaspberryPi Compute Module 4, RaspberryPi 400 – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.63.9.20220430-rpi4.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.63.9.20220430/RaspberryMatic-3.63.9.20220430-rpi4.zip)<br/>SHA256: d43ba267beeff7cd838f926d705c3b3440436066f5d1da52c79febc7cdcb805a</sub>

- [CCU3](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-CCU3), [ELV-Charly](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-ELV-Charly), RaspberryPi3 Model B+, RaspberryPi3 Model B, RaspberryPi3 Model A+, RaspberryPi Compute Module 3, RaspberryPi Compute Module 3 lite, RaspberryPi Zero 2 W – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.63.9.20220430-rpi3.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.63.9.20220430/RaspberryMatic-3.63.9.20220430-rpi3.zip)<br/>SHA256: 7b7dc00a9179fce47ff2f41a0565948bc61ff2dc906c98b17cbbdd0b7967b135</sub>
<sub>:package: [RaspberryMatic-3.63.9.20220430-ccu3.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.63.9.20220430/RaspberryMatic-3.63.9.20220430-ccu3.tgz) (only for initial CCU3 Firmware -> RaspberryMatic Upgrade)<br/> SHA256: 5bcde8060f396e7a3c72892b047836e359e8e5f2e6756766807ac99959fdf681</sub>

- RaspberryPi2 Model B – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.63.9.20220430-rpi2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.63.9.20220430/RaspberryMatic-3.63.9.20220430-rpi2.zip)<br/>SHA256: cefb9cb0c92cf925953e85e7967118a4b4823eb84a5978beb20eb7331d28835d</sub>

- RaspberryPi Zero W, RaspberryPi Zero, RaspberryPi Compute Module 1, RaspberryPi1 (A+/B+) – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.63.9.20220430-rpi0.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.63.9.20220430/RaspberryMatic-3.63.9.20220430-rpi0.zip)<br/>SHA256: 4ba5d467bfd1efb61bb758b1a2c8e4bfc157d59d5a6264a1c6176496db381807</sub>

- TinkerBoard S, TinkerBoard – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Tinkerboard)):
<sub>:package: [RaspberryMatic-3.63.9.20220430-tinkerboard.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.63.9.20220430/RaspberryMatic-3.63.9.20220430-tinkerboard.zip)<br/>SHA256: 1d4b5e98ddcba50bef6577943a132d5515136609350e257f09dcbaa75e54b632</sub>

- ODROID-N2/N2+/C4/C2 – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Odroid)):
<sub>:package: [RaspberryMatic-3.63.9.20220430-odroid-n2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.63.9.20220430/RaspberryMatic-3.63.9.20220430-odroid-n2.zip)<br/>SHA256: 9fcdca3fac0ad0ee0e24c9ae928e2b98168adff3fca1c2a738cf90535c55d8bb</sub>
<sub>:package: [RaspberryMatic-3.63.9.20220430-odroid-c4.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.63.9.20220430/RaspberryMatic-3.63.9.20220430-odroid-c4.zip)<br/>SHA256: 0033d10e9fa727f4a7ab2c317f73df185e66deade9bf5912a4514f3683ae0ecf</sub>
<sub>:package: [RaspberryMatic-3.63.9.20220430-odroid-c2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.63.9.20220430/RaspberryMatic-3.63.9.20220430-odroid-c2.zip)<br/>SHA256: 3b99c599fe0e07b6628acd001744561c01cf82f9c27c73aa654ada0fea464dd2</sub>

- Intel NUC – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-IntelNUC)):
<sub>:package: [RaspberryMatic-3.63.9.20220430-intelnuc.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.63.9.20220430/RaspberryMatic-3.63.9.20220430-intelnuc.zip)<br/>SHA256: ca084b2a5b74827ea1bb9610e64727a14932217947de70655faed35640fde280</sub>

- Open Virtual Appliance (OVA) – ([ProxmoxVE](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Proxmox-VE), [VirtualBox](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-VirtualBox), [ESXi](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-vmWare-ESXi), [Synology](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Synology-VMM), [QNAP](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-QNAP-VirtualizationStation), [Workstation Player](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-vmWare-Workstation-Player), [QEmu](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-QEmu), [UNRAID](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-UNRAID), [HyperV](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HyperV)):
<sub>:package: [RaspberryMatic-3.63.9.20220430-ova.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.63.9.20220430/RaspberryMatic-3.63.9.20220430-ova.zip)<br/>SHA256: f3fbfb76a771f4b88065c11221015267cec45a0c07c5e589fb173d5b25cf41b1</sub>
<sub>:package: [RaspberryMatic-3.63.9.20220430.ova](https://github.com/jens-maus/RaspberryMatic/releases/download/3.63.9.20220430/RaspberryMatic-3.63.9.20220430.ova) (only for initial OVA installation)<br/>SHA256: c107eff0ec6f4de49914f02821e449ce826452d9bd213fc1cc1eec25a5f32e77</sub>

- Docker / Open Container Initiative (OCI) – virtual appliance ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Docker-OCI)):
<sub>:package: [RaspberryMatic-3.63.9.20220430-oci_amd64.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.63.9.20220430/RaspberryMatic-3.63.9.20220430-oci_amd64.tgz) (amd64/x86_64)<br/>SHA256: 4b76cc091e5aeefd098bfe7ac998e628981c0e7ef302275cf2c5ac1af062a32c</sub>
<sub>:package: [RaspberryMatic-3.63.9.20220430-oci_arm64.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.63.9.20220430/RaspberryMatic-3.63.9.20220430-oci_arm64.tgz) (arm64/aarch64)<br/>SHA256: d7d76f9e771ac49c16d86d48452cf62633af76060a7d0b5a76abc98dbf7b26a3</sub>
<sub>:package: [RaspberryMatic-3.63.9.20220430-oci_arm.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.63.9.20220430/RaspberryMatic-3.63.9.20220430-oci_arm.tgz) (arm/armhf)<br/>SHA256: ad3f2c7df718e9b2d6091f388b92a9d814ba0e85f33a953d238bfb2e2af88517</sub>

- Kubernetes / K8s – virtual appliance:
<sub>see [install documentation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Kubernetes)</sub>

- Home Assistant Add-on – virtual appliance:
<sub>see [install documentation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant)</sub>

<hr /><em>This discussion was created from the release <a href='https://github.com/jens-maus/RaspberryMatic/releases/tag/3.63.9.20220430'>RaspberryMatic 3.63.9.20220430</a>.</em>

**Comment by oetiker** [comment]
Trying to upgrade my CCU3. Unfortunately this does not seem to be possible as it says: 1.8 GB need to be free. I looked around the device using the ssh login, but did not find anything big I could remove ... 

Any hints ?

![image](https://user-images.githubusercontent.com/429279/166420735-c79cc89f-8c2b-44ab-8db5-23986f02d3d5.png)

**Reply by jens-maus** [reply]
> Trying to upgrade my CCU3. Unfortunately this does not seem to be possible as it says: 1.8 GB need to be free. I looked around the device using the ssh login, but did not find anything big I could remove ...
> 
> Any hints ?

When you were switching over from the original CCU3 firmware to RaspberryMatic you were probably accidently skipping step 9 (Notwendigen System-Reset durchführen) of the official CCU3 installation documentation:

https://github.com/jens-maus/RaspberryMatic/wiki/Installation-CCU3 

So what you would need to do now is:

1. Create a fresh backup (*.sbk)
2. Perform a factory reset
3. Restore the backup

Afterwards the whole space of the standard SD card of a CCU3 will be available and thus an update should be possible again.

**Reply by oetiker** [reply]
Great! Now I have 5.6 GB free and the upgrade worked perfectly! Thank you!
